README.md

```diff
@@ -191,6 +191,120 @@ var reactHtml = ReactDOMServer.renderToStaticMarkup(
 assert.equal(reactHtml, htmlExpected);
 ```
 
+### Preprocessing element based on a criteria before rendering
+
+There may be a situation where you want to preprocess an element before rendering
+it via a custom component.
+This is beneficial if you want to add attributes to all the matching elements
+and then render them separately
+
+#### Example
+
+Below is a simple template in which you may want to replace ids of a div but also update the component.
+Replacing ids can be shared.
+
+##### Before
+```
+<div class="row">
+  <div id="first" data-process="shared">
+    <p>Sample For First</p>
+  </div>
+  <div id="second" data-process="shared">
+    <p>Sample For Second</p>
+  </div>
+</div>
+```
+
+##### After
+
+The ids are replaced during preprocessing and also the components are update via individual processors.
+
+```
+<div class="row">
+  <h1 id="preprocessed-first">First</h1>
+  <h2 id="preprocessed-second">Second</h2>
+</div>
+```
+
+#### Setup
+
+In your instructions object, you must specify `replaceChildren: true`.
```

Comment:
Why is `replaceChildren: true` necessary to use preprocessors?

Reply:
Missed it somehow. Fixed now.